app/src/main/java/br/org/cesar/discordtime/stickysessions/navigation/wrapper/ContextWrapper.java

```diff
@@ -0,0 +1,37 @@
+package br.org.cesar.discordtime.stickysessions.navigation.wrapper;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+
+import org.jetbrains.annotations.NotNull;
+
+import br.org.cesar.discordtime.stickysessions.navigation.repository.ActivityRepository;
+import br.org.cesar.discordtime.stickysessions.navigation.repository.IActivityRepository;
+
+
+public class ContextWrapper implements IContextWrapper {
+
+        @NotNull
+        private final Context mContext;
+        private final IActivityRepository mActivityRepository;
+
+        public void goNext(String activityName, boolean shouldClearStack) {
+            Class<? extends Activity> className =
+                    mActivityRepository.mapToClass(activityName);
+            Intent intent = new Intent(mContext, className);
+            if (shouldClearStack) {
+                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK|
+                        Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            }
+
+            mContext.startActivity(intent);
+        }
+
+        public ContextWrapper(@NotNull Context context) {
+            super();
```

Comment:
Why are we calling `super()`? ContextWrapper doesn't extend from anyone. super here means Object constructor.